src/components/diff/ImageDiff.tsx

```diff
@@ -0,0 +1,128 @@
+import { Contents } from '@jupyterlab/services';
+import { PromiseDelegate } from '@lumino/coreutils';
+import { Panel } from '@lumino/widgets';
+import { ReactWidget } from '@jupyterlab/apputils';
+
+import * as React from 'react';
+import ReactCompareImage from 'react-compare-image';
+
+import { Git } from '../../tokens';
+import {
+  ITranslator,
+  nullTranslator,
+  TranslationBundle
+} from '@jupyterlab/translation';
+import { Toolbar } from '@jupyterlab/apputils';
+
+export const createImageDiff: Git.Diff.ICallback = async (
+  model: Git.Diff.IModel,
+  toolbar?: Toolbar,
+  translator?: ITranslator
+): Promise<ImageDiff> => {
+  const widget = new ImageDiff(model, translator.load('jupyterlab_git'));
+  await widget.ready;
+  return widget;
+};
+
+type CompareImageProps = {
+  reference: string;
+  challenger: string;
+}
+
+const CompareImage = ({ reference, challenger }: CompareImageProps) => {
+  return (
+    <ReactCompareImage
+      leftImage={`data:image/png;base64,${reference}`}
+      rightImage={`data:image/png;base64,${challenger}`}
+      leftImageLabel="Reference"
+      rightImageLabel="Challenger"
+    />
+  )
+}
+
+export class ImageDiff extends Panel implements Git.Diff.IDiffWidget {
+  constructor(model: Git.Diff.IModel, translator?: TranslationBundle) {
+    super();
+    const getReady = new PromiseDelegate<void>();
+    this._container = this.node as HTMLElement;
+    this._isReady = getReady.promise;
+    this._model = model;
+    this._trans = translator ?? nullTranslator.load('jupyterlab_git');
+
```

Comment:
In here you can add a class to the container:

```typescript
this.addClass('jp-git-imagediff');
```

Then you could use it to set some CSS rules in a css file with the `style` folder.